mongorm/Document.py

```diff
@@ -43,11 +44,11 @@ def save( self, forceInsert=False, **kwargs ):
 				newId = collection.insert( self._data, **kwargs )
 			else:
 				newId = collection.save( self._data, **kwargs )
-		except pymongo.errors.OperationFailure, err:
+		except pymongo.errors.OperationFailure as err:
 			message = 'Could not save document (%s)'
-			if u'duplicate key' in unicode(err):
+			if u'duplicate key' in str(err):
```

Comment:
Of all the things Python 2.x. did horribly, strings are definitely at the top of the list! :P I would highly suggest including the `six` library and using its `six.text_type` here to save yourself further headaches.

Reply:
I ended up using `future` over `six` for this migration (just because it seems to better transform the code into something that's py3 compatible, and doesn't require you to do almost two rewrites (one that's six compatible, and one that's solely py3 compatible in the future)). And I used the `future` library with the ```from builtins import str``` at the top :D